odl/space/cu_ntuples.py

```diff
@@ -847,36 +976,271 @@ def _norm_default(x):
     return x.data.norm()
 
 
+def _pnorm_default(x, p):
+    if p == float('inf'):
+        raise NotImplementedError('inf-norm not implemented.')
+    elif int(p) != p:
+        raise NotImplementedError('non-integer powers not implemented')
+    # TODO: optimized version in C++ code?
+    xp = abs(x)
+    xp **= p
+    return sum(xp)**(1/p)
```

Comment:
Maybe, maybe not. It's confined to the module, so it doesn't really hurt much. And when you get an error, it will point to the `sum` function in the module, so it's also easy to debug. Anyway, not in this pull request.